File: c/linkedList.c
#include <stdio.h>
#include <stdlib.h>

// define node structure
struct Node {
  int data;
  struct Node* next;
};

// function to create a new node 
struct Node* createNode(int data) {
  struct Node* newNode = (struct Node*)malloc(sizeof(struct Node));
  newNode->data = data;
  newNode->next = NULL;
  return newNode;
}

// function to insert a node at beginning 
void insertAtBeginning(struct Node** head, int data) {
  struct Node* newNode = createNode(data);
  newNode->next = *head;
  *head = newNode;
}

// function to print linked list
void printList(struct Node* head) {
  struct Node* temp = head;
  while(temp != NULL) {
    printf("%d -> ", temp->data);
    temp = temp->next;
  }
  printf("NULL\n");
}

// main function to test linked list
int main() {
  struct Node* head = NULL;

  insertAtBeginning(&head, 10);
  insertAtBeginning(&head, 20);
  insertAtBeginning(&head, 30);

  printf("linked list: ");
  printList(head);
  return 0;
}

File: python/linked_list.py
class Node:
  def __init__(self, data):
    self.data = data
    self.next = None

class LinkedList:
  def __init__(self):
    self.head = None


  def insert_at_beginning(self, data):
    new_node = Node(data)
    new_node.next = self.head
    self.head = new_node

  def print_list(self):
    temp = self.head
    while temp:
      print(temp.data, end=" ->")
      temp = temp.next
    print("None")


# example usage
if __name__ == "__main__":
  l1 = LinkedList()
  l1.insert_at_beginning(10)
  l1.insert_at_beginning(20)
  l1.insert_at_beginning(30)

  print("linked list; ")
  l1.print_list()
